File: src/main/java/burrows/apps/example/gif/presentation/main/MainContract.java
package burrows.apps.example.gif.presentation.main;

import burrows.apps.example.gif.presentation.BasePresenter;
import burrows.apps.example.gif.presentation.BaseView;
import burrows.apps.example.gif.data.rest.model.RiffsyResponse;

/**
 * @author <a href="mailto:jaredsburrows@gmail.com">Jared Burrows</a>
 */
final class MainContract {
  // View
  interface View extends BaseView<Presenter> {
    void clearImages();
    void addImages(RiffsyResponse response);
    void showDialog(String url);
    boolean isActive();
  }

  // Presenter
  interface Presenter extends BasePresenter {
    void loadTrendingImages();
    void loadSearchImages(String searchString);
  }
}

File: src/main/java/burrows/apps/example/gif/presentation/main/MainContract.kt
package burrows.apps.example.gif.presentation.main

import burrows.apps.example.gif.data.model.RiffsyResponseDto
import burrows.apps.example.gif.BasePresenter
import burrows.apps.example.gif.BaseView
import burrows.apps.example.gif.presentation.adapter.model.ImageInfoModel

interface MainContract {
  interface Presenter : BasePresenter {
    fun clearImages()
    fun loadTrendingImages(next: Double?)
    fun loadSearchImages(searchString: String, next: Double?)
  }

  interface View : BaseView<Presenter> {
    fun clearImages()
    fun addImages(riffsyResponseDto: RiffsyResponseDto)
    fun showDialog(imageInfoModel: ImageInfoModel)
    fun isActive(): Boolean
  }
}
